src/realm-settings/ExecutorForm.tsx

```diff
@@ -55,54 +59,123 @@ export default function ExecutorForm() {
     ClientProfileRepresentation[]
   >([]);
   const [profiles, setProfiles] = useState<ClientProfileRepresentation[]>([]);
-  const form = useForm<ExecutorForm>({ defaultValues });
-  const { control } = form;
+  const form = useForm({ defaultValues });
+  const { control, setValue, handleSubmit } = form;
+  const editMode = executorName ? true : false;
 
   useFetch(
     () =>
       adminClient.clientPolicies.listProfiles({ includeGlobalProfiles: true }),
     (profiles) => {
       setGlobalProfiles(profiles.globalProfiles ?? []);
       setProfiles(profiles.profiles ?? []);
+
+      const profile = profiles.profiles!.find(
+        (profile) => profile.name === profileName
+      );
+
+      const profileExecutor = profile?.executors!.find(
+        (executor) => executor.executor === executorName
+      );
+
+      if (profileExecutor) {
+        Object.entries(profileExecutor).map(([key, value]) => {
+          if (key === "configuration") {
+            setValue("config", value);
+          }
+        });
+      }
     },
     []
   );
 
-  const fldNameFormatter = (name: string) =>
-    name.toLowerCase().trim().split(/\s+/).join("-");
-
   const save = async () => {
     const formValues = form.getValues();
     const updatedProfiles = profiles.map((profile) => {
       if (profile.name !== profileName) {
         return profile;
       }
 
+      const profileExecutor = profile.executors!.find(
+        (executor) => executor.executor === executorName
+      );
+
       const executors = (profile.executors ?? []).concat({
         executor: formValues.executor,
-        configuration: formValues.configuration,
+        configuration: formValues.config,
       });
 
-      return {
-        ...profile,
-        executors,
-      };
+      let editedExecutorConfig = {};
+      if (editMode) {
+        editedExecutorConfig = Object.assign(
+          profileExecutor!.configuration,
+          formValues.config
+        );
+        profileExecutor!.configuration = editedExecutorConfig;
+      }
+
+      if (editMode) {
+        return {
+          ...profile,
+        };
```

Comment:
Is there a need to spread this value? Perhaps it can be returned immediately?
```suggestion
        return profile;
```